.changeset/khaki-meals-prove.md

```diff
@@ -0,0 +1,9 @@
+---
+"@remix-run/architect": minor
+"@remix-run/cloudflare": minor
+"@remix-run/node": minor
+"@remix-run/react": minor
+"@remix-run/server-runtime": minor
+---
+
+feat: deprecate all functions & types exported from `magicExports`
```

Comment:
Note: we don't need to use conventional commit formatting for these. They should generally read as plain sentences, and they'll be grouped by type in the release notes automatically by Changesets.

```suggestion
Importing functions and types from the `remix` package is deprecated, and all exported modules will be removed in the next major release. For more details, [see the release notes for 1.4.0](https://github.com/remix-run/remix/releases/tag/v1.4.0) where these changes were first announced.
```